src/Symfony/Bridge/ProxyManager/Tests/LazyProxy/PhpDumper/ProxyDumperTest.php

```diff
@@ -80,6 +80,37 @@ public function testGetProxyFactoryCode()
         );
     }
 
+    /**
+     * @dataProvider getPrivatePublicDefinitions
+     */
+    public function testCorrectPublicPrivateInstantiation(Definition $definition, string $access)
+    {
+        $definition->setLazy(true);
+
+        $code = $this->dumper->getProxyFactoryCode($definition, 'foo', '$this->getFoo2Service(false)');
+
+        $this->assertStringMatchesFormat(
+            '%A$this->'.$access.'[\'foo\'] = %A',
+            $code
+        );
+    }
+
+    public function getPrivatePublicDefinitions()
```

Comment:
make static?

Reply:
what for? it's phpunit dataprovider

Reply:
https://phpunit.de/manual/6.5/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers

Reply:
https://github.com/sebastianbergmann/phpunit/blob/master/src/Util/Test.php#L840

using $this inside the data provider can be confusing if it's not static too, use of Prophecy for example isn't managed etc...  it was just a minor suggestion.

Reply:
But nowhere in the Symfony tests data proiders are made static. Why exactly must this one be?